sunpy/net/base_client.py

```diff
@@ -426,22 +425,14 @@ def fetch(self, *query_results, path=None, overwrite=False, progress=True,
             Results to download.
         path : `str` or `pathlib.Path`, optional
             Path to the download directory
-        overwrite : `bool`, optional
-            Replace files with the same name if True.
-        progress : `bool`, optional
-            Print progress info to terminal.
-        max_conns : `int`, optional
-            Maximum number of download connections.
-        downloader : `parfive.Downloader`, optional
+        downloader : `parfive.Downloader`
             The download manager to use.
-        wait : `bool`, optional
-           If `False` ``downloader.download()`` will not be called. Only has
-           any effect if `downloader` is not `None`.
 
         Returns
         -------
         `parfive.Results`
-            The results object, can be `None` if ``wait`` is `False`.
+            The results object, can be `None` if ``wait`` is `False` and
```

Comment:
What is wait for?

Reply:
Documenting this is a split brain problem, wait is to provide a way for Fido to tell old clients which can be used on their own to behave in a way that's Fido compatible i.e. not call `downloader.download`

Reply:
Do still want to do this? Is there not more we can cut?

Reply:
Well deprecating the old API to JSOC, VSO and dataretriever would be nice but a lot of work and deprecations

Reply:
Sure but feels like something we should at least look into?